adapters/orbidder/orbidder_test.go

```diff
@@ -32,3 +50,116 @@ func TestJsonSamples(t *testing.T) {
 
 	adapterstest.RunJSONBidderTest(t, "orbiddertest", bidder)
 }
+
+var testCasesCurrency = map[string]struct {
+	imp         openrtb2.Imp
+	setMock     func(m *mock.Mock)
+	expectedImp openrtb2.Imp
+}{
+	"EUR: no bidfloor, no currency": {
+		imp: openrtb2.Imp{
+			BidFloor:    0,
+			BidFloorCur: "",
+		},
+		setMock: func(m *mock.Mock) {},
+		expectedImp: openrtb2.Imp{
+			BidFloor:    0,
+			BidFloorCur: "EUR",
+		},
+	},
+	"EUR: bidfloor, no currency": {
+		imp: openrtb2.Imp{
+			BidFloor:    1,
+			BidFloorCur: "",
+		},
+		setMock: func(m *mock.Mock) {},
+		expectedImp: openrtb2.Imp{
+			BidFloor:    1,
+			BidFloorCur: "EUR",
+		},
+	},
+	"EUR: bidfloor and currency": {
+		imp: openrtb2.Imp{
+			BidFloor:    1,
+			BidFloorCur: "EUR",
+		},
+		setMock: func(m *mock.Mock) {},
+		expectedImp: openrtb2.Imp{
+			BidFloor:    1,
+			BidFloorCur: "EUR",
+		},
+	},
+	"USD: bidfloor with currency": {
+		imp: openrtb2.Imp{
+			BidFloor:    1,
+			BidFloorCur: "USD",
+		},
+		setMock: func(m *mock.Mock) {
+			m.On("GetRate", "USD", "EUR").Return(2.5, nil)
+		},
+		expectedImp: openrtb2.Imp{
+			BidFloor:    2.5,
+			BidFloorCur: "EUR",
+		},
+	},
+	"USD: no bidfloor": {
+		imp: openrtb2.Imp{
+			BidFloor:    0,
+			BidFloorCur: "USD",
+		},
+		setMock: func(m *mock.Mock) {},
+		expectedImp: openrtb2.Imp{
+			BidFloor:    0,
+			BidFloorCur: "EUR",
+		},
+	},
+}
+
+var testCasesExtension = map[string]struct {
+	imp       openrtb2.Imp
+	expectErr bool
+}{
+	"Valid Orbidder Extension": {
+		imp: openrtb2.Imp{
+			Ext: json.RawMessage(`{"bidder":{"accountId":"orbidder-test", "placementId":"center-banner", "bidfloor": 0.1}}`),
+		},
+		expectErr: false,
+	},
+	"Invalid Orbidder Extension": {
+		imp: openrtb2.Imp{
+			Ext: json.RawMessage(`{"there's'":{"something":"strange", "in the":"neighbourhood", "who you gonna call?": 0.1}}`),
+		},
+		expectErr: true,
+	},
+}
+
+func TestPreprocessBidFloorCurrency(t *testing.T) {
+	for name, tc := range testCasesCurrency {
+		t.Run(name, func(t *testing.T) {
+			imp := tc.imp
+			mockConversions := &mockCurrencyConversion{}
+			tc.setMock(&mockConversions.Mock)
+			extraRequestInfo := adapters.ExtraRequestInfo{
+				CurrencyConversions: mockConversions,
+			}
+
+			err := preprocessBidFloorCurrency(&imp, &extraRequestInfo)
```

Comment:
You need to call `mockCurrencyConversion.AssertExpectations(t)` to make sure that the calls that you're expecting to the currency converter implementation are called with the args you have specified in `setMock` in those test cases.

Reply:
Thank you for pointing this out to us. We added a call of `AsserrExpectations` in our test method. Always happy to learn new stuff :)